mrp_lock_lot/models/mrp_production_lot.py

```diff
@@ -21,7 +21,14 @@ class StockProductionLot(models.Model):
 
     @api.one
     def _get_locked_value(self):
-        return self.product_id.categ_id.lot_default_locked
+        '''Lock by default?'''
+        categ = self.product_id.categ_id
```

Comment:
Better to have a computed field `locked` that makes this ascent through parent categories and use it in all the places.